tls/tls_srv.c

```diff
@@ -560,7 +560,7 @@ ttls_choose_ciphersuite(TlsCtx *tls)
 			ci = ttls_ciphersuite_from_id(ciphersuites[i]);
 			if (!ci) {
 				TTLS_WARN(tls, "ClientHello: cannot match a ciphersuite\n");
-				return T_BLOCK;
+				return T_BAD;
```

Comment:
Please pay attention. May be you have idea about more appropriate error code from errno.h